pyClient/zeth/utils.py

```diff
@@ -232,8 +233,11 @@ def encode_to_hash(message_list: Any) -> bytes:
         m_hex = m
 
         # Convert it into a hex
-        if isinstance(m, int):
-            m_hex = "{0:0>4X}".format(m)
+        if isinstance(m, (FQ, int)):
+            assert int(m) < 2**256, "The input cannot be encoded as " + \
+                "a bytes32 (longer than 256 bits)"
```

Comment:
Why assert here? If you field element is > 2^{256} (ie: the bit-size of an Ethereum word, please specify that in a comment, otherwise we don't understand why you `assert` here), you can use multiple limbs to represent it, and so use as many bytes32 as necessary to represent your field element @rrtoledo

Reply:
I wrote the assert as well as the comment before  ( ` BEWARE //!\\ We rely on the fact that the prime (q of Fq) < 2**256-1` ) only to warn us in case we change of Fp.

We could use multiple limbs for sure. I am not certain of what we would do however to process the primary inputs in the contracts (we could also process them as bytes[] I guess) which is why I left it open and wrote this comment and assert as a warning/reminder.

Reply:
Added comment (256 being the bit-size of an Ethereum word )

Reply:
> we could also process them as bytes[] I guess

yes, a field element would now be encoded on several Ethereum words (several bit-strings of length 256)